src/modules/data/views/Messages/MessageDetails.tsx

```diff
@@ -13,146 +13,51 @@
 import { useDataTranslation } from '../../registration';
 
-interface Props {
-  message: IMessage;
-  open: boolean;
-  onClose: () => void;
-}
-
-export const MessageDetails: React.FC<Props> = ({ message, open, onClose }) => {
+export const MessageDetails: () => JSX.Element = () => {
+  const history = useHistory();
   const { t } = useDataTranslation();
+  const { id } = useParams<{ id: string }>();
   const classes = useStyles();
-  const history = useHistory();
-  const [loading, setLoading] = useState(false);
-  const [txId, setTxId] = useState<string>();
-  const [data, setData] = useState<IData[]>([]);
   const { selectedNamespace } = useContext(NamespaceContext);
-  const { setMessage, setMessageType } = useContext(SnackbarContext);
-
-  const detailItem = (label: string, value: string | JSX.Element) => (
-    <>
-      <Grid item xs={12}>
-        <Typography className={classes.detailLabel}>{label}</Typography>
-      </Grid>
-      <Grid item xs={12}>
-        {value}
-      </Grid>
-    </>
-  );
+  const [loading, setLoading] = useState(false);
+  const [message, setMessage] = useState<IMessage>();
 
   useEffect(() => {
     setLoading(true);
-    Promise.all([
-      fetchWithCredentials(
-        `/api/v1/namespaces/${selectedNamespace}/batches/${message.batch}`
-      ),
-      fetchWithCredentials(
-        `/api/v1/namespaces/${selectedNamespace}/messages/${message.header.id}?data`
-      ),
-    ])
-      .then(async ([batchResponse, messageDataResponse]) => {
-        if (batchResponse.ok && messageDataResponse.ok) {
-          setData((await messageDataResponse.json()).data);
-          const batch: IBatch = await batchResponse.json();
-          setTxId(batch.payload.tx.id);
+    fetchWithCredentials(
+      `/api/v1/namespaces/${selectedNamespace}/messages/${id}`
+    )
+      .then(async (messageResponse) => {
+        if (messageResponse.ok) {
+          setMessage(await messageResponse.json());
         } else {
-          setMessageType('error');
-          setMessage(`Error loading details for message ${message.header.id}`);
+          console.log('error fetching message');
         }
       })
       .finally(() => {
         setLoading(false);
       });
-  }, [
-    message.batch,
-    selectedNamespace,
-    message.header.id,
-    setMessage,
-    setMessageType,
-  ]);
-
-  const copyableHash = (hash: string) => (
-    <Grid alignItems="center" container direction="row">
-      <Grid item xs={8}>
-        <Typography
-          title={hash}
-          noWrap
-          className={clsx(classes.detailValue, classes.paddingRight)}
-        >
-          {hash}
-        </Typography>
-      </Grid>
-      <Grid item xs={4}>
-        <CopyToClipboard text={hash}>
-          <Button size="small" className={classes.copyButton}>
-            {t('copy')}
-          </Button>
-        </CopyToClipboard>
-      </Grid>
-    </Grid>
-  );
-
-  const transactionLink = (txId: string | undefined) => (
```

Comment:
I think it's important for us to provide a link to the transaction the message was in, if there was one.

Reply:
Added:
![image](https://user-images.githubusercontent.com/17093033/145499474-ca61961a-bc70-47e7-9041-b77903c7a8a3.png)

Reply:
On this one, can we also show the transaction hash (if there is one) in a popover?

Reply:
https://user-images.githubusercontent.com/17093033/145601385-dee35e46-e596-4ec9-aa58-7683a7ba9886.mov

Reply:
https://user-images.githubusercontent.com/17093033/145601508-c94359ac-a62a-4a3e-b686-719e39ac4354.mov